dataprofiler/profilers/profile_builder.py

```diff
@@ -1160,7 +1160,10 @@ def __init__(self, data, samples_per_update=None, min_true_samples=0,
         self.row_has_null_count = 0
         self.row_is_null_count = 0
         self.hashed_row_dict = dict()
-        self._profile = dict()
+        self._profile = []
+        self._col_name_to_idx = dict()
+        self._initialized = False
+        self._duplicate_cols_present = False
```

Comment:
we can make these static values, but the operations themselves could just be reactive properties to information we already store so we don't have to store more info but rather have it as a property function.